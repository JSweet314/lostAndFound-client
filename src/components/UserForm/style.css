.user-form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 10rem 0 0 0;
  background-color: rgba(0, 0, 0, 0.5);
  height: 100%;
}

.user-form__input,
.user-form__btn,
.password-fields__password-toggle-btn,
.user-form__password-fields {
  width: 50%;
  margin: 1rem;
  background-color: rgba(0, 0, 0, 0.5);
  border: 2px solid #fff;
  font-family: 'Exo', sans-serif;
  font-size: 2rem;
  color: #fff;
}

.user-form__greeting {
  border: none;
  margin: 1rem;
  font-size: 4rem;
  text-align: center;
  color: #fff;
}

.password-fields__password-toggle-btn,
.user-form__btn,
.user-form__input {
  padding: 1rem;
  color: #fff;
}

.password-fields__input::placeholder,
.user-form__input::placeholder {
  font-family: 'Exo', sans-serif;
  color: rgba(256, 256, 256, 0.5);
}
.password-fields__password-toggle-btn,
.user-form__btn {
  border: 2px solid #625049;
  box-shadow: 2px 2px 10px #C28744
}

.password-fields__password-toggle-btn:hover,
.user-form__btn:hover {
  background-color: #C28744;
  color: #fff;
}

.user-form__btn:disabled {
  color: rgba(256, 256, 256, 0.5);
  box-shadow: none;
}

.user-form__btn:disabled:hover {
  background-color: #C28744;
  color: #fff;
  cursor: default;
}

.user-form__password-fields {
  display: flex;
}

.password-fields__input {
  background-color: rgba(0, 0, 0, 0.5);
  width: 100%;
  font-size: 2rem;
  padding: 1rem;
  border: none;
  color: #fff;
}

.password-fields__password-toggle-btn {
  padding: 0 1rem;
  margin: 0;
  border: none;
  width: 12rem;
  box-shadow: none;
}