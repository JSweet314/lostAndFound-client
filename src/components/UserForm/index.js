import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

const UserForm = ({
  routeId, 
  username, 
  email, 
  password, 
  handleOnChange, 
  handleOnSubmit, 
  showPassword,
  togglePasswordVisibility,
  errorMessage
}) => {
  const welcomeBackDisplay = routeId === 'login' ? 'initial' : 'none';
  const usernameDisplay = routeId === 'login' ? 'none' : 'initial';
  const passwordDisplayBtnText = showPassword ? 'hide' : 'show';
  const buttonText = routeId === 'login' ? 'Log In' : 'Sign Up';
  const userNameRequired = routeId === 'login' ? false : true;
  const showDisabled = password ? false : true;
  const isDisabled = email && password ? false : true;
  const type = showPassword ? 'text' : 'password';

  return (
    <form
      className='user-form'
      onSubmit={event => handleOnSubmit(event)}>
      <input 
        className='user-form__input'
        style={{display: usernameDisplay}}
        onChange={event => handleOnChange(event)}
        value={username}
        name='username'
        placeholder="name"
        required={userNameRequired}
        type='text' />
      <h2  
        className='user-form__greeting' 
        style={{display: welcomeBackDisplay}}>
        Welcome Back!
      </h2>
      <input 
        className='user-form__input'
        onChange={event => handleOnChange(event)}
        value={email}
        name='email'
        placeholder='email'
        required
        type='email' />
      <div className='user-form__password-fields'>
        <input 
          className='password-fields__input'
          onChange={event => handleOnChange(event)}
          value={password}
          name='password'
          placeholder='password'
          required
          type={type} />
        <button 
          disabled={showDisabled}
          className='password-fields__password-toggle-btn'
          onClick={event => togglePasswordVisibility(event)}>
          {passwordDisplayBtnText}
        </button>
      </div>
      {
        errorMessage && 
        <h1 className='error'>
          Invalid username/password, try again or sign up
        </h1>
      }
      <button 
        className='user-form__btn user-form__btn--submit'
        disabled={isDisabled}
        type='submit'>
        {buttonText}
      </button>
    </form>
  );
};

UserForm.propTypes = {
  routeId: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  showPassword: PropTypes.bool.isRequired,
  handleOnChange: PropTypes.func.isRequired,
  handleOnSubmit: PropTypes.func.isRequired,
  togglePasswordVisibility: PropTypes.func.isRequired,
  errorMessage: PropTypes.string
};

export default UserForm;