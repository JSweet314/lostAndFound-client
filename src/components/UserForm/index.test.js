import React from 'react';
import {shallow} from 'enzyme';
import UserForm from './index';

describe('UserForm', () => {
  let wrapper;
  const mockHandleOnChange = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <UserForm 
        name=''
        email=''
        password=''
        routeId="login"
        handleOnChange={mockHandleOnChange}
      />
    );
  });
  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});