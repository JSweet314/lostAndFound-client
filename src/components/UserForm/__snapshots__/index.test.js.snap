// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserForm
    email=""
    handleOnChange={[Function]}
    name=""
    password=""
    routeId="login"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="name"
          onChange={[Function]}
          placeholder="name"
          style={
                    Object {
                              "display": "none",
                            }
          }
          type="text"
          value=""
/>,
        <input
          name="email"
          onChange={[Function]}
          placeholder="email"
          type="email"
          value=""
/>,
        <input
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="password"
          value=""
/>,
        <button
          type="submit"
>
          Log In
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "name",
          "onChange": [Function],
          "placeholder": "name",
          "style": Object {
            "display": "none",
          },
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "email",
          "onChange": [Function],
          "placeholder": "email",
          "type": "email",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "password",
          "onChange": [Function],
          "placeholder": "password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Log In",
          "type": "submit",
        },
        "ref": null,
        "rendered": "Log In",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="name"
            onChange={[Function]}
            placeholder="name"
            style={
                        Object {
                                    "display": "none",
                                  }
            }
            type="text"
            value=""
/>,
          <input
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="email"
            value=""
/>,
          <input
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
/>,
          <button
            type="submit"
>
            Log In
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "name",
            "onChange": [Function],
            "placeholder": "name",
            "style": Object {
              "display": "none",
            },
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "email",
            "onChange": [Function],
            "placeholder": "email",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "password",
            "onChange": [Function],
            "placeholder": "password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Log In",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Log In",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
